File: src/gestorAplicacion/operacional/Producto.java
package gestorAplicacion.operacional;

 import java.io.Serializable;
import java.util.ArrayList;
 
public class Producto implements Serializable {  
	
	//-----------SERIALIZADOR-----------
	private static final long serialVersionUID = 1L;
	
	private static ArrayList<Producto> productos;
	static {
		productos = new ArrayList<Producto>();
	}
	
	public static ArrayList<Producto> getProductos() {
		return productos;
	}
	
	public static void setProductos(ArrayList<Producto> productos) {
		Producto.productos = productos;
	}
	
	
	//-----------ATRIBUTOS-----------
	public static int numProducto = 0;
	private int productoId;
	private String nombreProducto;
	private float precioVenta;
	
	//-----------CONSTRUCTOR-----------
	public Producto(String nombreProducto, float precioVenta){
		this.productoId = Producto.numProducto;
		//this.existencias = existencias;
		this.nombreProducto = nombreProducto;
		this.precioVenta = precioVenta;
		Producto.numProducto++;
		productos.add(this);
	}
	
	//-----------GETTERS y SETTERS-----------
	public int getProductoId() {
		return productoId;
	}
	public void setProductoId(int productoId) {
		this.productoId = productoId;
	}
	
	public String getNombreProducto() {
		return nombreProducto;
	}
	public void setNombreProducto(String nombreProducto) {
		this.nombreProducto = nombreProducto;
	}
	public float getPrecioVenta() {
		return precioVenta;
	}
	public void setPrecioVenta(float precioVenta) {
		this.precioVenta = precioVenta;
	}
	
	//-----------OTROS METODOS-----------
	public String toString() {
		return "El producto= "+ nombreProducto +" "+ "tiene un precio de venta de = " + precioVenta;
		
	}

	
}

File: Python/src/gestorAplicacion/operacional/Producto.py
class Producto:
    
 
    #-----------ATRIBUTOS-----------
    numProducto = 0
    _producto = []     #serializador

    #-----------CONSTRUCTOR-----------
    def __init__(self, nombreProducto, precioVenta):
    
        self._productoId = 0
        self._nombreProducto = None
        self._precioVenta = 0

        self._productoId = Producto.numProducto
        #this.existencias = existencias
        self._nombreProducto = nombreProducto
        self._precioVenta = precioVenta
        Producto.numProducto += 1
        Producto._producto.append(self)     #serializador

    #-----------GETTERS y SETTERS-----------

    @classmethod          #serializador
    def getProductos(cls):
        return cls._producto

    @classmethod             #serializador
    def setProductos(cls, productos):
        cls._producto = productos

    def getProductoId(self):
        return self._productoId
    def setProductoId(self, productoId):
        self._productoId = productoId

    def getNombreProducto(self):
        return self._nombreProducto
    def setNombreProducto(self, nombreProducto):
        self._nombreProducto = nombreProducto
    def getPrecioVenta(self):
        return self._precioVenta
    def setPrecioVenta(self, precioVenta):
        self._precioVenta = precioVenta

    #-----------OTROS METODOS-----------
    def __str__(self):
        return "El producto= "+ self._nombreProducto +" "+ "tiene un precio de venta de = " + str(self._precioVenta)



